modules/math/src/main/java/com/opengamma/strata/math/impl/interpolation/LinearInterpolator1D.java

```diff
@@ -34,56 +32,51 @@
 
   @Override
   public double interpolate(Interpolator1DDataBundle data, double value) {
-    JodaBeanUtils.notNull(data, "data");
-    InterpolationBoundedValues boundedValues = data.getBoundedValues(value);
-    double x1 = boundedValues.getLowerBoundKey();
-    double y1 = boundedValues.getLowerBoundValue();
-    if (data.getLowerBoundIndex(value) == data.size() - 1) {
+    int lowerIndex = data.getLowerBoundIndex(value);
+    double y1 = data.getValue(lowerIndex);
+    if (lowerIndex == data.size() - 1) {
       return y1;
     }
-    double x2 = boundedValues.getHigherBoundKey();
-    double y2 = boundedValues.getHigherBoundValue();
+    double x1 = data.getKey(lowerIndex);
+    double x2 = data.getKey(lowerIndex + 1);
+    double y2 = data.getValue(lowerIndex + 1);
     return y1 + (value - x1) / (x2 - x1) * (y2 - y1);
   }
 
   @Override
   public double firstDerivative(Interpolator1DDataBundle data, double value) {
-    JodaBeanUtils.notNull(data, "data");
-    InterpolationBoundedValues boundedValues = data.getBoundedValues(value);
-    double x1 = boundedValues.getLowerBoundKey();
-    double y1 = boundedValues.getLowerBoundValue();
-    if (data.getLowerBoundIndex(value) == data.size() - 1) {
+    int lowerIndex = data.getLowerBoundIndex(value);
+    if (lowerIndex == data.size() - 1) {
+      if (data.size() == 1) {
```

Comment:
Yes, but it would be slower. Given a typical curve has more than one node, and a typical lookup is not after the last node, it is better to catch both cases with a single if, rather than two.